components/canvas/Canvas.tsx

```diff
@@ -58,10 +60,27 @@ const Canvas = ({
     new Date("2024-04-24T00:08:00.000000Z").getTime();
 
   let tempNodes: Node<NodeDataFull>[] = [];
-  let tempEdges: Edge[] = apiEdges;
+  let tempEdges: Edge[] = [];
+  apiEdges.map((edge: EdgeDataApi) => {
+    const nodeSource = apiNodes.filter((node) => node.id === edge.source);
+    if (nodeSource[0] && nodeSource[0].type === NodeTypes.choice) {
+      tempEdges.push({
+        ...edge,
+        type: "choice",
+        label: edge.edgeValue,
+      });
+    } else {
+      tempEdges.push({ ...edge });
+    }
+  });
+
   const [nodes, setNodes, onNodesChange] = useNodesState<NodeDataFull>([]);
   const [edges, setEdges, onEdgesChange] = useEdgesState([]);
 
+  const edgeTypes = {
+    choice: ChoiceEdge,
+  };
```

Comment:
Edge types should be in a seperate file or we will get console warnings. I made a file for this in my branch. If you copy it and  replace the edge type thats there with yours then it will be easy for me to merge. https://github.com/equinor/flyt/blob/621-feat-deletable-edges/components/canvas/EdgeElementTypes.tsx

Reply:
done